GameProject/Assets/Turn based assets/Scripts/MouseSelection.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using UnityEngine;
+
+namespace Turn_based_assets.Scripts
+{
+    public class MouseSelection : MonoBehaviour
+    {
+        private ISelection _selection;
+        [SerializeField] private GameObject cube;
+        public float distanceY;
+        public float offset;
+        public float gridSize;
+        private Plane _plane;
+        private Vector3 distanceFromCamera;
+        
+
+        private void Start()
+        {
+            var position = Camera.main.transform.position;
+            distanceFromCamera = new Vector3(position.x, position.y - distanceY,position.z);
+            _plane = new Plane(Vector3.up, distanceFromCamera);
+        }
+
+        private void Update()
+        {
+            Ray ray = Camera.main.ScreenPointToRay(Input.mousePosition);
+
+            float gridSpace = 0;
+
+            if (_plane.Raycast(ray, out gridSpace))
```

Comment:
Were not in C# 4 - this can simply be `if (_plane.Raycast(ray, out float gridSpace))` or `if (_plane.Raycast(ray, out var gridSpace))`, with myself preferring the former in this case as it is a numerical datatype.